clients/drcachesim/tracer/raw2trace.h

```diff
@@ -103,6 +147,47 @@ struct instr_summary_t final {
     {
         return next_pc_;
     }
+    /** Get the pc of the start of the block containing this instrucion. */
+    app_pc
+    tag() const
+    {
+        return tag_;
+    }
+    /** Get the pc of the start of this instrucion. */
+    app_pc
+    pc() const
+    {
+        return pc_;
+    }
+
+    /**
+     * Sets properties of the "pos"-th source memory operand by OR-ing in the
+     * two boolean values.
+     */
+    void
+    set_mem_src_flags(size_t pos, bool use_remembered_base, bool remember_base)
+    {
+        DEBUG_ASSERT(pos < mem_srcs_and_dests_.size());
+        mem_srcs_and_dests_[pos].use_remembered_base =
+            mem_srcs_and_dests_[pos].use_remembered_base || use_remembered_base;
+        mem_srcs_and_dests_[pos].remember_base =
+            mem_srcs_and_dests_[pos].remember_base || remember_base;
+    }
+
+    /**
+     * Sets properties of the "pos"-th destination memory operand by OR-ing in the
+     * two boolean values.
+     */
+    void
+    set_mem_dest_flags(size_t pos, bool use_remembered_base, bool remember_base)
+    {
+        DEBUG_ASSERT(num_mem_srcs_ + pos < mem_srcs_and_dests_.size());
+        mem_srcs_and_dests_[num_mem_srcs_ + pos].use_remembered_base =
```

Comment:
+1.  Done.  (Since this is C++, technically `|=` should be safe: it just doesn't feel right though and I'd be worried somebody would then model some of our C code on it...)